jlibra-examples/src/main/java/dev/jlibra/example/AsyncTransferExample.java

```diff
@@ -0,0 +1,143 @@
+            U64Argument amountArgument = new U64Argument(1 * 1000000);
+            AccountAddressArgument addressArgument = new AccountAddressArgument(target.getByteSequence());
+            Transaction transaction = ImmutableTransaction.builder()
+                    .sequenceNumber(i)
+                    .maxGasAmount(140000)
+                    .gasUnitPrice(0)
+                    .senderAccount(AccountAddress.ofPublicKey(publicKeySource))
+                    .expirationTime(Instant.now().getEpochSecond() + 60)
+                    .payload(ImmutableScript.builder()
+                            .code(Move.peerToPeerTransferAsBytes())
+                            .addArguments(addressArgument, amountArgument)
+                            .build())
+                    .build();
+            SignedTransaction signedTransaction = ImmutableSignedTransaction.builder()
+                    .publicKey(publicKeySource)
+                    .transaction(transaction)
+                    .signature(ImmutableSignature.builder()
+                            .privateKey(privateKeySource)
+                            .transaction(transaction)
+                            .build())
+                    .build();
+            transactions.add(admissionControl
+                    .asyncSubmitTransaction(signedTransaction));
+            logger.info("Created transaction {}, sending 1 libra from {} to {}", i, source.getByteSequence(),
+                    target.getByteSequence());
+        }
+
+        logger.info("All transactions created. Wait for them to be accepted..");
+        long time = System.currentTimeMillis();
+        CompletableFuture.allOf(transactions.toArray(new CompletableFuture[transactions.size()])).get();
+        logger.info("Transactions accepted in {} ms, get account states (both accouns should have 10 Libras): ",
+                System.currentTimeMillis() - time);
+
+        // sleep to make sure the transactions have been executed (submitting them
+        // successfully only means they are accepted for execution). The correct way
+        // would be to poll the api to make sure the transactions have been executed.
+        Thread.sleep(2000);
+
+        CompletableFuture<UpdateToLatestLedgerResult> result = admissionControl
+                .asyncUpdateToLatestLedger(
+                        ImmutableQuery.builder()
+                                .accountStateQueries(asList(
+                                        ImmutableGetAccountState.builder()
+                                                .address(target)
+                                                .build(),
+                                        ImmutableGetAccountState.builder()
+                                                .address(source)
+                                                .build()))
+                                .build());
+
+        result.get().getAccountResources().forEach(accountResource -> {
```

Comment:
If you save `CompletableFuture.allOf` result into a variable instead calling `get()` you could chain it to create one async flow:

```
CompletableFuture<Void> asyncExample = allSubmitsCombined
                .thenCompose(aVoid -> result)
                .thenApply(UpdateToLatestLedgerResult::getAccountResources)
                .thenAccept(resources -> resources
                        .forEach(accountResource -> {
                            logger.info("-------------------------------------------------------------------");
                            logger.info("Authentication key: {}", accountResource.getAuthenticationKey());
                            logger.info("Received events: {}", accountResource.getReceivedEvents().getCount());
                            logger.info("Sent events: {}", accountResource.getSentEvents().getCount());
                            logger.info("Balance (microLibras): {}", accountResource.getBalanceInMicroLibras());
                            logger.info("Balance (Libras): {}", new BigDecimal(accountResource.getBalanceInMicroLibras())
                                    .divide(BigDecimal.valueOf(1_000_000)));
                            logger.info("Sequence number: {}", accountResource.getSequenceNumber());
                        }));

        // to trigger async execution
        asyncExample.get();

        channel.shutdown();
    }
```